src/components/TransactionsTabContent/TransactionsTabContent.tsx

```diff
@@ -31,8 +33,12 @@ export const TransactionsTabContent = ({ transactions, TxDialog, EditDialog }: T
         ) : (
           <div className="flex flex-col items-center gap-1 text-center">
             <h3 className="text-2xl font-bold tracking-tight">You have no Transactions</h3>
-            <p className="text-sm text-muted-foreground mb-2">You can record Transaction here.</p>
-            <TxDialog />
+            {showRecordTransaction && (
+              <>
+                <p className="text-sm text-muted-foreground mb-2">You can record Transaction here.</p>
+                <TxDialog />
```

Comment:
Make `<TxDialog />` optional in `TransactionsTabContentProps` interface. Then check if `TxDialog` is defined.

Reply:
Why?

Reply:
Passing TxDialog as a prop will define if TxDialog should be rendered.

Reply:
And how would it make it clearer for reader or easier to maintain when comparing to flag with a documentation?

Reply:
Passing props it's a common practice in React projects. One of the use cases is a conditional rendering. This is what we have here. If you want to use flag - no problem

Reply:
I know that it's common practice, but the question was how does it make the intention of the code clearer.
The flag with documentation why it's hidden for me it's clear from the first look. While, if I would look for the first time, and see some optional parameter, which isn't passed at all, would either think that we should remove this optional parameter or I would ask a question to my self WTF.